### PR TITLE
Document change to size > 0 setting for request cache

### PR DESCRIPTION
**Pull request by peteralfonsi**
### Description
Minor change: The setting allowing caching size > 0 queries, from https://github.com/opensearch-project/OpenSearch/pull/16484, will be tweaked in https://github.com/opensearch-project/OpenSearch/pull/16570. This PR changes its documentation. 

Because the original setting never entered any release other than the 2.x branch I think it's fine to change its name, but correct me if I'm wrong. 

### Issues Resolved
N/A

### Version
2.x (new change in open source)

### Frontend features
N/A

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by peteralfonsi** [comment]
Also, @kolchfa-aws I saw the backport 2.18 label is added - the original setting isn't in 2.18, just 2.x, so I think the backport shouldn't be needed.

**Comment by kolchfa-aws** [comment]
@peteralfonsi What do you mean by 2.x? Which versions of OpenSearch is this applicable to?

File: _search-plugins/caching/request-cache.md [hunk 1]
```diff
@@ -28,7 +28,7 @@ Setting | Data type  | Default | Level | Static/Dynamic | Description
 `indices.cache.cleanup_interval` | Time unit  | `1m` (1 minute)  | Cluster | Static | Schedules a recurring background task that cleans up expired entries from the cache at the specified interval. 
 `indices.requests.cache.size` | Percentage | `1%`      | Cluster | Static | The cache size as a percentage of the heap size (for example, to use 1% of the heap, specify `1%`). 
 `index.requests.cache.enable` | Boolean    | `true`    | Index | Dynamic | Enables or disables the request cache. 
-`indices.requests.cache.enable_for_all_requests` | Boolean    | `false`    | Cluster | Dynamic | Enables or disables caching queries in which `size` is greater than `0`.
+`indices.requests.cache.maximum_cacheable_size` | Integer    | `0`    | Cluster | Dynamic | Sets the maximum `size` for queries to be added to the request cache.
```
Comment (natebower) [inline _search-plugins/caching/request-cache.md:31]:
```suggestion
`indices.requests.cache.maximum_cacheable_size` | Integer    | `0`    | Cluster | Dynamic | Sets the maximum `size` of queries to be added to the request cache.
```

**Review by natebower — COMMENTED** [review]
@kolchfa-aws One minor change. Thanks!

**Pull request merged**